x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -79,119 +73,15 @@ private static void fieldLevelMappingIssue(IndexMetaData indexMetaData, BiConsum
-            List<String> issues = new ArrayList<>();
-                Map<String, Settings> filters = indexMetaData.getSettings().getGroups(AnalysisRegistry.INDEX_ANALYSIS_FILTER);
-                for (Map.Entry<String, Settings> entry : filters.entrySet()) {
-                if ("delimited_payload_filter".equals(entry.getValue().get("type"))) {
-                    issues.add("The filter [" + entry.getKey() + "] is of deprecated 'delimited_payload_filter' type. "
-                            + "The filter type should be changed to 'delimited_payload'.");
-                }
-                }
-            if (issues.size() > 0) {
-                return new DeprecationIssue(DeprecationIssue.Level.WARNING,
-                        "Use of 'delimited_payload_filter'.",
-                        "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking_70_analysis_changes.html",
-                        issues.toString());
-            }
-        }
-        return null;
-    }
-
-    static DeprecationIssue indexStoreTypeCheck(IndexMetaData indexMetaData) {
-        if (indexMetaData.getCreationVersion().before(Version.V_6_0_0_alpha1) &&
-            indexMetaData.getSettings().get("index.store.type") != null) {
-            return new DeprecationIssue(DeprecationIssue.Level.CRITICAL,
-                "The default index.store.type has been removed. If you were using it, " +
-                    "we advise that you simply remove it from your index settings and Elasticsearch" +
-                    "will use the best store implementation for your operating system.",
-                "https://www.elastic.co/guide/en/elasticsearch/reference/master/" +
-                    "breaking_60_settings_changes.html#_store_settings", null);
-
-        }
-        return null;
-    }
-
-    static DeprecationIssue storeThrottleSettingsCheck(IndexMetaData indexMetaData) {
-        if (indexMetaData.getCreationVersion().before(Version.V_6_0_0_alpha1)) {
-            Settings settings = indexMetaData.getSettings();
-            Settings throttleSettings = settings.getAsSettings("index.store.throttle");
-            ArrayList<String> foundSettings = new ArrayList<>();
-            if (throttleSettings.get("max_bytes_per_sec") != null) {
-                foundSettings.add("index.store.throttle.max_bytes_per_sec");
-            }
-            if (throttleSettings.get("type") != null) {
-                foundSettings.add("index.store.throttle.type");
-            }
-
-            if (foundSettings.isEmpty() == false) {
+    static DeprecationIssue oldIndicesCheck(IndexMetaData indexMetaData) {
+        Version createdWith = indexMetaData.getCreationVersion();
+        if (createdWith.before(Version.V_7_0_0)) {
                 return new DeprecationIssue(DeprecationIssue.Level.CRITICAL,
-                    "index.store.throttle settings are no longer recognized. these settings should be removed",
+                    "Index created before 7.0",
```

Comment:
Can you help me understand why indices created prior to 7.0 is a critical (or even an issue), and why are we linking to the 8.0 breaking changes. 

The code looks fine, i think i am just missing something here.

Reply:
Absolutely: Our policy as listed in [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html) is that each Elasticsearch version supports indices created in the previous major version, but any that are older than that can't be used.  That is, 6.x can open indices created in 5.x, but not indices created in 2.x - those must be reindexed, either in 5.x before upgrading or using reindex-from-remote.

Due to that, the Deprecations API reports any indices which need to be reindexed before upgrading to the next major version: in 6.x, we'll flag indices which were created in 5.x and need to be reindexed before upgrading the cluster to 7.x.

With this PR, the Deprecations API on 7.x clusters will report any indices that will need to be reindexed before upgrading to the next major version (assumed to be 8.0). This lets us leave at least one check in place as an example, even though there aren't any breaking changes in 8.0 yet (because 8.0 doesn't exist yet).

We may want to switch out the link, as it is kind of making assumptions about the structure of documentation that doesn't exist yet - I could make it point to the page listing the policy about indices created in previous versions for now.

Reply:
Thanks for the explains, that makes sense! I think guessing the 8.0 URL is a safe bet and plenty of time to fix it guessed wrong.